File: app/src/main/java/com/example/madarsofttask/application/BaseApplication.java
package com.example.madarsofttask.application;

import android.app.Application;

import androidx.room.Room;

import com.example.madarsofttask.data.db.UserDatabase;

public class BaseApplication extends Application {
    public static UserDatabase userDatabase;

    @Override
    public void onCreate() {
        super.onCreate();
        initDB();
    }

    private void initDB() {
        userDatabase = Room.databaseBuilder(getApplicationContext(), UserDatabase.class, "userDB").allowMainThreadQueries().build();
    }
}

File: app/src/main/java/com/example/madarsofttask/application/BaseApplication.kt
package com.example.madarsofttask.application

import android.app.Application
import androidx.room.Room
import com.example.madarsofttask.data.db.UserDatabase

class BaseApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        initDB()
    }

    private fun initDB() {
        userDatabase = Room.databaseBuilder(applicationContext, UserDatabase::class.java, "userDB").allowMainThreadQueries().build()
    }

    companion object {
        @JvmField
        var userDatabase: UserDatabase? = null
    }
}